nautobot/core/ui/object_detail.py

```diff
@@ -820,24 +818,15 @@ def get_extra_context(self, context: Context):
             if self.order_by_fields:
                 body_content_table_queryset = body_content_table_queryset.order_by(*self.order_by_fields)
             body_content_table_queryset = body_content_table_queryset.distinct()
+            body_content_table = body_content_table_class(
+                body_content_table_queryset, hide_hierarchy_ui=self.hide_hierarchy_ui
+            )
             if self.tab_id:
                 # Use the `self.tab_id`, if it exists, to determine the correct return URL for the table
                 # to redirect the user back to the correct tab after editing/deleteing an object
-                body_content_table = body_content_table_class(
-                    body_content_table_queryset,
-                    hide_hierarchy_ui=self.hide_hierarchy_ui,
-                    extra_columns=[
-                        (
-                            "actions",
-                            ButtonsColumn(
-                                model=body_content_table_queryset.model, return_url_extra=f"?tab={self.tab_id}"
-                            ),
-                        )
-                    ],
-                )
-            else:
-                body_content_table = body_content_table_class(
-                    body_content_table_queryset, hide_hierarchy_ui=self.hide_hierarchy_ui
+                body_content_table.base_columns["actions"] = ButtonsColumn(
+                    model=body_content_table_queryset.model,
+                    return_url_extra=f"?tab={self.tab_id}",
```

Comment:
This still feels off to me - lots of tables have a custom `actions` column (e.g. with nonstandard `buttons` tuple, custom `prepend_template`, etc.) that we're just overwriting here with something generic. Can we modify the existing object with a `return_url_extra` rather than replacing it entirely? Or something like that?

Reply:
Should be fixed in this commit https://github.com/nautobot/nautobot/pull/7158/commits/be37636e0a6d9b2c21a6ce778452641d44321a0f